lib/init.coffee

```diff
@@ -10,4 +13,32 @@ module.exports =
       type: 'string'
 
   activate: ->
-    console.log 'activate linter-puppet-lint'
+    @regex = '(?<type>(warning|error)): (?<message>.+?) on line (?<line>\\d+) col (?<col>\\d+)'
+
+    @subscriptions = new CompositeDisposable
+    @subscriptions.add atom.config.observe 'linter-puppet-lint.puppetLintExecutablePath',
+      (executablePath) =>
+        @executablePath = executablePath
+    @subscriptions.add atom.config.observe 'linter-puppet-lint.puppetLintArguments',
+      (args) =>
+        @args = [ "--log-format", "'%{kind}: %{message} on line %{line} col %{column}'" ]
+        @args = @args.concat args.split(' ')
+  deactivate: ->
+    @subscriptions.dispose()
+
+  puppetLinter: ->
+    provider =
+      grammarScopes: ['source.puppet']
+      scope: 'file'
+      lintOnFly: false
+      lint: (textEditor) =>
+        args = @args[..]
+        args.push textEditor.getPath()
+
+        helpers.exec(@executablePath, args)
+          .then (val) =>
+            return helpers.parse(val, @regex)
```

Comment:
You might wanna make this `return helpers.parse(val, @regex, filePath: textEditor.getPath())`